PegSolitaire.cpp

```diff
@@ -0,0 +1,207 @@
+﻿
+#include "PegSolitaire.hpp"
+
+const std::array<std::array<char, 9>, 9> ps::get_english_board() {
+	std::fstream file("../PegSolitaireBoard.txt");
+	std::array<std::array<char, 9>, 9> board = {};
+	if (file.is_open()) {
+		std::cout << "Reading file" << std::endl;
+		std::string line;
+		int row = 0;
+		while (std::getline(file, line)) {
+			auto s = line.size();
+			for (int col = 0; col < line.size(); col++) {
+				auto charValue = line[col];
+				if (charValue == 'E') {
+					return board;
+				}
+				else if (charValue == '1') {
+					board[row][col] = '1';
+				}
+				else if (charValue == '0') {
+					board[row][col] = '0';
+				}
+				else {
+					board[row][col] = ' ';
+				}
+			}
+			row++;
+		}
+	}
+	else {
+		std::cerr << "Error opening file" << std::endl;
+	}
+	return board;
+}
+void ps::print(const std::array<std::array<char, 9>, 9>* board) {
+	for (auto& line : *board) {
+		for (auto& chr : line) {
+			std::cout << chr;
+		}
+		std::cout << std::endl;
+	}
+}
+void updatePath(std::vector<std::array<char, 5>>* path, const char lastDirection,
+	const char currentRow, const char currentCol, const char rowOffset, const char colOffset) {
+	std::array<char, 5> temp = {};
+	temp[0] = lastDirection;
+	temp[1] = currentRow;
+	temp[2] = currentCol;
+	temp[3] = currentRow + rowOffset;
+	temp[4] = currentCol + colOffset;
+	path->push_back(temp);
+}
+static bool check_X_away(const std::array<std::array<char, 9>, 9>* board, const char currentRow, const char currentCol,
+	const char rowOffset, const char colOffset, const char checkValue) {
+	if ((*board)[currentRow + rowOffset][currentCol + colOffset] == checkValue) {
+		return true;
+	}
+	return false;
+}
+static char remainingPieces(const std::array<std::array<char, 9>, 9>* board) {
+	char remainingPieces = 0;
+	for (auto& line : *board) {
+		for (auto& chr : line) {
+			if (chr == '1') {
+				++remainingPieces;
+				if (remainingPieces > 1) {
+					return remainingPieces;			// Return even if you don't count them all
+				}
+			}
+		}
+	}
+	return remainingPieces;
+}
+/// <summary>
+/// FOR rowOffset AND colOffset USE ONLY VALUE -2, 0, +2 CORESPONDING TO AXIS DIRECTION
+/// </summary>
+static void makePegJump(std::array<std::array<char, 9>, 9>* board, const char currentRow, const char currentCol,
+	const char rowOffset, const char colOffset, const char* setBits) {
+	(*board)[currentRow][currentCol] = setBits[0];								  // PEACE MAKING A JUMP		//setBits_normal -> '0'
+	(*board)[currentRow + rowOffset / 2][currentCol + colOffset / 2] = setBits[1];// PEACE THAT IS JUMPED OVER	//setBits_normal -> '0'
+	(*board)[currentRow + rowOffset][currentCol + colOffset] = setBits[2];		  // PEACE NEW POSITION			//setBits_normal -> '1'
+}

```

Comment:
What is the purpose of `setBits`?
Why not just
```suggestion
static void makePegJump(std::array<std::array<char, 9>, 9>* board, const char currentRow, const char currentCol,
	const char rowOffset, const char colOffset) {
	(*board)[currentRow][currentCol] = '0';								  // PEACE MAKING A JUMP		//setBits_normal -> '0'
	(*board)[currentRow + rowOffset / 2][currentCol + colOffset / 2] =  '0';// PEACE THAT IS JUMPED OVER	//setBits_normal -> '0'
	(*board)[currentRow + rowOffset][currentCol + colOffset] = '1';		  // PEACE NEW POSITION			//setBits_normal -> '1'
}

```

Reply:
I used char setBits_normal[3] = { '0', '0', '1' }; and char setBits_reverse[3] = { '1', '1', '0' }; with idea of reversing the board back by actual play back instead of hashing board state. It is not used anymore and can be simplified now.